File: 백준/Silver/14382. 숫자세는 양 （Large）/숫자세는 양 （Large）.kt
import java.io.BufferedReader
import java.io.File
import java.io.InputStreamReader
import java.util.*

fun main() {
    val os = System.getProperty("os.name").lowercase(Locale.getDefault())
    val args:List<String> = if (os == "linux") {
        val br = BufferedReader(InputStreamReader(System.`in`))
        br.readLines()
    } else {
        val inputFile = File("src/main/kotlin/input.txt")
        inputFile.readLines()
    }
    val numbers = args.slice(1 until args.size).map{ it.toLong()}
    val answer = mutableListOf<String>()

    for (number in numbers) {
        if (number == 0L) {
            answer.add("INSOMNIA")
            continue
        }

        val set = mutableSetOf<Char>()
        for (j in 1..1000000L) {
            (number * j).toString().forEach { set.add(it) }
            if (set.size == 10) {
                answer.add((number * j).toString())
                break
            }
        }
    }

    answer.forEachIndexed { idx, el ->
        println("Case #${idx + 1}: $el")
    }
}

File: 백준/Silver/14382. 숫자세는 양 （Large）/숫자세는 양 （Large）.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
let input = fs.readFileSync(filePath)
    .toString()
    .trim()
    .split("\n")
    .map(x => x.trim())

function solution([n, ...args]) {
    const numbers = args.map(Number);
    let answer = [];
    for (let i = 0; i < numbers.length; i++) {
        const number = numbers[i];
        if (number === 0) {
            answer.push('INSOMNIA');
            continue;
        }
        const set = new Set();
        for (let j = 1; j <= 10 ** 6; j++) {
            [...(number * j).toString()].forEach(numStr => set.add(numStr));
            if (set.size === 10) {
                answer.push(number * j);
                break;
            }
        }
    }
    answer = answer.map((el, idx) => `Case #${idx + 1}: ${el}`);
    console.log(answer.join('\n'));
}


solution(input);
